File: src/types/export.ts
export const PROJECT_EXPORT_FORMAT_VALUES = ['pdf'] as const

export type ProjectExportFormat = (typeof PROJECT_EXPORT_FORMAT_VALUES)[number]

export const PROJECT_EXPORT_STATUS_VALUES = ['pending', 'running', 'completed', 'failed'] as const

export type ProjectExportStatus = (typeof PROJECT_EXPORT_STATUS_VALUES)[number]

export interface ProjectExportTask {
  artifact_name: string | null
  artifact_size_bytes: number | null
  completed_pages: number
  created_at: string
  current_page_number: number | null
  current_page_title: string | null
  download_url: string | null
  error: string | null
  format: ProjectExportFormat
  id: string
  progress: number
  project_id: string
  stage: string
  status: ProjectExportStatus
  total_pages: number
  updated_at: string
}

File: python-backend/app/schemas/export.py
from __future__ import annotations

from datetime import datetime
from enum import StrEnum

from pydantic import Field

from app.schemas.base import APIModel


class ProjectExportFormat(StrEnum):
    PDF = "pdf"


class ProjectExportStatus(StrEnum):
    PENDING = "pending"
    RUNNING = "running"
    COMPLETED = "completed"
    FAILED = "failed"


class ProjectExportTaskResponse(APIModel):
    id: str
    project_id: str
    format: ProjectExportFormat
    status: ProjectExportStatus
    stage: str
    total_pages: int = Field(ge=0)
    completed_pages: int = Field(ge=0)
    current_page_number: int | None = Field(default=None, ge=1)
    current_page_title: str | None = None
    progress: float = Field(default=0.0, ge=0.0, le=1.0)
    error: str | None = None
    artifact_name: str | None = None
    artifact_size_bytes: int | None = Field(default=None, ge=0)
    download_url: str | None = None
    created_at: datetime
    updated_at: datetime
